independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanRegistrar.java

```diff
@@ -34,6 +35,14 @@
             return configure(DotName.createSimple(beanClass.getName()));
         }
 
+        /**
+         * The synthetic observer is not added to the deployment unless the {@link ObserverConfigurator#done()} method is
+         * called.
+         * 
+         * @return a new synthetic observer configurator
+         */
+        ObserverConfigurator configureObserver();
```

Comment:
I am thinking this might deserve an `ObserverRegistrar` instead of merging it into `BeanRegistrar`.

Reply:
Well, it would be more correct and so but in fact, I'd rather rename the `BeanRegistrar` to `ComponentRegistrar`... but this would require some compatibility tweaks.